include/beman/scope/scope.hpp

```diff
@@ -6,9 +6,11 @@
 #include <type_traits>
 #include <utility>
 
+// clang-format off
 #if __cplusplus < 202002L
   #error "C++20 or later is required"
 #endif
+// clang-format on
```

Comment:
this doesn't work on MSVC if `/Zc:__cplusplus` is not passed as a compiler argument, we need to also check `_MSVC_LANG`.
Are we targetting MSVC?

Reply:
Then MSVC is non-compliant with the standard.  __cplusplus is the standard macro to check the language version -- if they fail to set that properly then that's a bug report to them.

Also for the moment we have no MSVC target in CI -- we need an actual implementation (PR on another branch) --instead of just exposing the TS -- before we can support MSVC.